src/ElmHtml/InternalTypes.elm

```diff
@@ -50,49 +51,58 @@ type alias TextTagRecord =
 
 {-| A node contains the `tag` as a string, the children, the facts (e.g attributes) and descendantsCount
 -}
-type alias NodeRecord =
+type alias NodeRecord msg =
     { tag : String
-    , children : List ElmHtml
+    , children : List (ElmHtml msg)
     , facts :
-        Facts
+        Facts msg
         --, namespace : String
     , descendantsCount : Int
     }
 
 
 {-| A markdown node contains facts (e.g attributes) and the model used by markdown
 -}
-type alias MarkdownNodeRecord =
-    { facts : Facts
+type alias MarkdownNodeRecord msg =
+    { facts : Facts msg
     , model : MarkdownModel
     }
 
 
 {-| Custom nodes contain facts (e.g attributes) and a json value for the model
 -}
-type alias CustomNodeRecord =
-    { facts : Facts
+type alias CustomNodeRecord msg =
+    { facts : Facts msg
     , model : Json.Decode.Value
     }
 
 
 {-| Tagger holds the map function when Html.Map is used, the tagger
-should then be applied to events comming from descendant nodes
+should then be applied to events comming from descendant nodes, it
+is basically a javascript function.
 -}
 type alias Tagger =
     Json.Decode.Value
 
 
+{-| EventHandler holds the function that is called when an event is
+triggered, it is basically a javascript object like this:
+
+{ decoder: [Function] }
+-}
+type alias EventHandler =
+    Json.Decode.Value
```

Comment:
although `Tagger` and `EventHandler` are just aliases to `Json.Decode.Value`, I like to have those alias so people can differentiate then when reading functions signatures, also, we can add comments to it that explains what values they hold, to help with the decoding.